geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java

```diff
@@ -100,6 +101,8 @@
     DataSerializableFixedID, Extensible<Region<?, ?>>, EvictableRegion {
 
   private static final Logger logger = LogService.getLogger();
+  final ReentrantReadWriteLock readWriteLockForCacheLoader = new ReentrantReadWriteLock();
```

Comment:
Shouldn't these be private? If they are accessed from another class, then a better approach would be either of these:

1) add getters -- doesn't really improve it a lot but at least we aren't reaching into another class to access and use fat locks.

2) make the uses of these locks completely contained within this class. If you really need to inject code from another class to run within a block holding these locks, then try something like:

```
  public static void execute(final Runnable action) {
    // acquire lock
    try {
      action.run();
    } finally {
      // release lock
    }
  }
```